### PR TITLE
update lottery ticket pruner based on refactored compression code

### PR DESCRIPTION
**Pull request by QuanluZhang**
1. update lottery ticket pruner based on refactored compression code
2. fix bug: allow users to specify device of the model when exporting mask and model, used to place dummy input tensor for exporting onnx model.

### CODE REVIEW

File: src/sdk/pynni/nni/compression/torch/compressor.py [hunk 1]
```diff
@@ -341,6 +347,21 @@ def export_model(self, model_path, mask_path=None, onnx_path=None, input_shape=N
...
+        Parameters:
+        -----------
+        model_state : dict
+            state dict saved from unwrapped model
+        """
+        if self.is_wrapped:
+            self._unwrap_model()
+            self.bound_model.load_state_dict(model_state)
+            self._wrap_model()
+        else:
```
Comment (Cjkkkk) [inline src/sdk/pynni/nni/compression/torch/compressor.py]: when will `is_wrapped`  be false ? Before calling `compress` method?

File: src/sdk/pynni/nni/compression/torch/compressor.py [hunk 2]
```diff
@@ -291,7 +294,10 @@ def _wrap_modules(self, layer, config):
             the configuration for generating the mask
         """
         _logger.info("compressing module %s.", layer.name)
-        return PrunerModuleWrapper(layer.module, layer.name, layer.type, config, self)
+        wrapper = PrunerModuleWrapper(layer.module, layer.name, layer.type, config, self)
+        assert hasattr(layer.module, 'weight')
```
Comment (Cjkkkk) [inline src/sdk/pynni/nni/compression/torch/compressor.py]: why not using `check_weight` method?

File: src/sdk/pynni/nni/compression/torch/compressor.py [hunk 3]
```diff
@@ -291,7 +294,10 @@ def _wrap_modules(self, layer, config):
...
+        wrapper.to(layer.module.weight.device)
```
Comment (Cjkkkk) [inline src/sdk/pynni/nni/compression/torch/compressor.py]: can you explain this?

File: src/sdk/pynni/nni/compression/torch/compressor.py (hunk 1 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/compressor.py]: yes.
it is possible that users call `load_model_state` before `compress`, we need an approach to check whether the model has been wrapped or not. any suggestion?

File: src/sdk/pynni/nni/compression/torch/compressor.py (hunk 2 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/compressor.py]: where is this method defined, and how to use it?

File: src/sdk/pynni/nni/compression/torch/compressor.py (hunk 3 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/compressor.py]: if the module is placed on GPU, buffers in the wrapped module are put on CPU, thus, I need to move the whole wrapper to the device that the module is placed

File: src/sdk/pynni/nni/compression/torch/compressor.py (hunk 2 above)
Comment (Cjkkkk) [reply src/sdk/pynni/nni/compression/torch/compressor.py]: https://github.com/microsoft/nni/blob/d452a1665f802bfbe75372b6c942272b252c70a2/src/sdk/pynni/nni/compression/torch/compressor.py#L563
